src/encoder.rs

```diff
@@ -1901,9 +1903,12 @@ fn encode_partition_topdown<T: Pixel>(
       if split_vert { partition_types.push(PartitionType::PARTITION_VERT); };
     }
     else {
-      //partition_types.append(&mut RAV1E_PARTITION_TYPES.to_vec());
-      partition_types.push(PartitionType::PARTITION_NONE);
-      partition_types.push(PartitionType::PARTITION_SPLIT);
+      if bsize.width_log2() == fi.min_partition_size.width_log2() + 1 {
+        partition_types.append(&mut RAV1E_PARTITION_TYPES.to_vec());
```

Comment:
```suggestion
        partition_types.extend_from_slice(RAV1E_PARTITION_TYPES);
```

This would avoid what is effectively an extra clone.

Reply:
Okay- thanks for the tip!